Gems/Terrain/Code/Source/TerrainSystem/TerrainSystem.cpp

```diff
@@ -177,6 +177,199 @@ bool TerrainSystem::InWorldBounds(float x, float y) const
     return false;
 }
 
+// Generate positions to be queried based on the sampler type.
+void TerrainSystem::GenerateQueryPositions(const AZStd::span<AZ::Vector3>& inPositions,
+    AZStd::vector<AZ::Vector3>& outPositions,
+    Sampler sampler, size_t indexStepSize) const
+{
+    const float minHeight = m_currentSettings.m_worldBounds.GetMin().GetZ();
+    for (size_t i = 0, iteratorIndex = 0; i < inPositions.size(); i++, iteratorIndex += indexStepSize)
```

Comment:
suggestion: If you use "outPositions.reserve()" instead of "outPositions.resize()" where outPositions gets defined, you could just make this loop "for (auto& inPoition : inPositions)" and use emplace_back() for pushing back each outPosition value.

Reply:
Fixed